Strata/DL/Lambda/LExprEval.lean

```diff
@@ -42,16 +42,19 @@ def isCanonicalValue (e : (LExpr LMonoTy IDMeta)) : Bool :=
     -- So we could simplify the following to `closed e`, but leave it as is for
     -- clarity.
     LExpr.closed e
-  | .mdata _ e' => isCanonicalValue e'
-  | _ => false
+  | .mdata _ e' => isCanonicalValue σ e'
+  | _ =>
+    match Factory.callOfLFunc σ.config.factory e with
+    | some (_, args, f) => f.isConstr && List.all (args.map (isCanonicalValue σ)) id
+    | none => false
 
 /--
 Equality of canonical values `e1` and `e2`.
 
 We can tolerate nested metadata here.
 -/
-def eql (e1 e2 : (LExpr LMonoTy IDMeta))
-  (_h1 : isCanonicalValue e1) (_h2 : isCanonicalValue e2) : Bool :=
+def eql (σ : (LState IDMeta)) (e1 e2 : (LExpr LMonoTy IDMeta))
```

Comment:
```suggestion
def eql (σ : LState IDMeta) (e1 e2 : LExpr LMonoTy IDMeta)
```
I general, if you change a line of code, you can also improve its legibility, I always appreciate it when there are less parentheses when it's not ambiguous